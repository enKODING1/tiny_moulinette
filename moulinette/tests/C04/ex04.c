#include <unistd.h>

void ft_putnbr_base(int nbr, char *base);

int main(void)
{
    ft_putnbr_base(4242, "");
    ft_putnbr_base(4242, "0");
    ft_putnbr_base(4242, "01");
    ft_putnbr_base(4242, "012345678+");
    ft_putnbr_base(4242, "012345678-");
    ft_putnbr_base(4242, "0123456788");
    ft_putnbr_base(4242, "0123455789");
    write(1, "\n---\n", 5);
    ft_putnbr_base(4242, "abcdefghij");
    write(1, "\n---\n", 5);
    ft_putnbr_base(-4242, "");
    ft_putnbr_base(-4242, "0");
    ft_putnbr_base(-4242, "01");
    ft_putnbr_base(-4242, "012345678+");
    ft_putnbr_base(-4242, "012345678-");
    write(1, "\n---\n", 5);
    ft_putnbr_base(-4242, "abcdefghij");
    write(1, "\n---\n", 5);
    ft_putnbr_base(0, "abcdefghij");
    write(1, "\n---\n", 5);
    ft_putnbr_base(4242, "ab");
    write(1, "\n---\n", 5);
    ft_putnbr_base(4242, "abcd");
    write(1, "\n---\n", 5);
    ft_putnbr_base(4242, "abcdefgh");
    write(1, "\n---\n", 5);
    ft_putnbr_base(4242, "abcdefghijklmnop");
    write(1, "\n---\n", 5);
    return 0;
}
