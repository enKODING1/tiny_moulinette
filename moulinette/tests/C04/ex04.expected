1000010010010
---
ecec
---
-1000010010010
---
-ecec
---
a
---
baaaabaabaaba
---
baacbac
---
baccc
---
bajc
---
